tasks/check-data-keys/check-data-keys.yaml

```diff
@@ -24,52 +29,52 @@ spec:
       description: The workspace where the data JSON file resides
   steps:
     - name: check-data-keys
-      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      # TODO: Update the image to the latest release-service-utils image
+      image: quay.io/redhat-user-workloads/rhtap-release-2-tenant/release-service/release-service-utils:on-pr-403edb0827b86a46933608edc09a8b8598373f0a 
       args: ["$(params.systems[*])"]
       script: |
         #!/usr/bin/env bash
         set -ex
 
-        KEYS_JSON='{
-            "releaseNotes": [
-                "product_id",
-                "product_name",
-                "product_version",
-                "product_stream",
-                "cpe",
-                "type",
-                "content.images",
-                "synopsis",
-                "topic",
-                "description",
-                "solution",
-                "references"
-            ],
-            "cdn": [
-                "env"
-            ]
-        }'
+        if [ ! -f "/workspace/data/$(params.dataPath)" ]; then
+            echo "Data JSON file not found: /workspace/data/$(params.dataPath)"
+            exit 1
+        fi
+
+        if ! curl -s "$(params.schema)" -o /tmp/schema.json; then
+            echo "Error: Failed to fetch schema from $(params.schema)"
+            exit 1
+        fi
 
-        if [ ! -f "$(workspaces.data.path)/$(params.dataPath)" ] ; then
-            echo "No data JSON was provided."
+        if ! jq empty /tmp/schema.json; then
+            echo "Error: Invalid JSON schema in /tmp/schema.json"
             exit 1
         fi
 
         RC=0
+        for system in "$@"; do
+          echo "Checking if system $system is supported"
+          if [[ $(jq -r ".properties | has(\"$system\")" /tmp/schema.json) != true ]]; then
+              echo "Unsupported system value: $system"
+              RC=1
+          fi
+        done
+
+        if [ $RC -ne 0 ]; then
+            exit 1
+        fi
 
-        for SYSTEM in "$@" ; do
-            if [[ $(jq -r ".$SYSTEM" <<< "$KEYS_JSON") == null ]] ; then
-                echo "Unsupported system value: $SYSTEM"
-                RC=1
-            fi
+        echo "Adding required fields to root for system: $*"
```

Comment:
I have also kept the old branch so we can discuss it.

Reply:
Closing the loop here, Sean and I met about it and as long as his json schema can do the same things as the current code, the current code can be replaced. So, the new solution has to check the systems passed, type check the values in the system, and ensure that the required fields are present in the system. It seems Sean got this working with some if when stuff in the schema

Reply:
OK, thanks for the update. I just looked briefly now, so let me see if my understanding is correct: The schema now has a new field `systems` which is not really part of the data file. But when the check-data-keys task is run, it will inject it there and that will make some of the field required and then the schema is validated. Is that how it works?

It seems a bit hacky, but fine I guess.

Reply:
That's how I understand it, yeah. It would be nice to not have the systems key in the schema at all, but it gets us everything else we want AFAICT. If we discover an improvement later on, great, but for now I think it is plenty good enough